src/wallet/wallet2.cpp

```diff
@@ -8569,7 +8569,8 @@ wallet2::request_stake_unlock_result wallet2::can_request_stake_unlock(const cry
         return result;
       }
 
-      if (contribution.amount < service_nodes::SMALL_CONTRIBUTOR_THRESHOLD && (curr_height - node_info.registration_height) < service_nodes::SMALL_CONTRIBUTOR_UNLOCK_TIMER)
+      uint64_t small_contributor_amount_threshold = mul128_div64(service_nodes::SMALL_CONTRIBUTOR_THRESHOLD, service_nodes::get_staking_requirement(nettype(), curr_height), oxen::STAKING_REQUIREMENT);
+      if (contribution.amount < small_contributor_amount_threshold && (curr_height - node_info.registration_height) < service_nodes::SMALL_CONTRIBUTOR_UNLOCK_TIMER)
```

Comment:
The math here is a bit more complex than needed: we're storing the threshold as OXEN, but we don't really mean OXEN, we want a numerator out of 15k.  It would be cleaner to declare it directly as a ratio, something like this:

```C++
using SMALL_CONTRIBUTOR_THRESHOLD = std::ratio<2499, 10000>;
```
and then:
```C++
uint64_t small_contributor_amount_threshold = mul128_div64(
    service_nodes::get_staking_requirement(nettype(), curr_height),
    SMALL_CONTRIBUTOR_THRESHOLD::num,
    SMALL_CONTRIBUTOR_THRESHOLD::denom);
```